### PR TITLE
Add the ability to edit pipeline config + schedule post pipeline deploy

### PR DESCRIPTION
**Pull request by ajainarayanan**

### CODE REVIEW

File: cdap-ui/app/directives/my-pipeline-configurations/my-realtime-pipeline-config/my-realtime-pipeline-config-ctrl.js [hunk 1]
```diff
@@ -151,9 +196,120 @@ class MyRealtimePipelineConfigCtrl {
...
+    )
+      .$promise
+      .then(
+        () => {
+          this.$state.reload();
+        },
+        (err) => {
+          this.myAlertOnValium.show({
+            type: 'danger',
+            content: typeof err === 'object' ? JSON.stringify(err) : 'Updating pipeline failed: '+ err
```
Comment (elfenheart) [inline cdap-ui/app/directives/my-pipeline-configurations/my-realtime-pipeline-config/my-realtime-pipeline-config-ctrl.js]: nit: space before `+`

File: cdap-ui/app/directives/my-pipeline-configurations/my-realtime-pipeline-config/my-realtime-pipeline-config.html [hunk 2]
```diff
@@ -14,7 +14,355 @@
...
+        </div>
+        <div class="label-with-toggle checkpointing row">
+          <span class="toggle-label col-xs-4">Checkpointing</span>
+          <div class="col-xs-7 toggle-container">
+            <my-toggle-switch-widget
+              is-on="!RealtimePipelineConfigCtrl.checkpointing"
+              on-toggle="RealtimePipelineConfigCtrl.onCheckPointingChange()"
+            ></my-toggle-switch-widget>
+            <i class="fa fa-info-circle"
+                uib-tooltip="Allows Apache Spark Streaming to checkpoint data (RDDs) to persistent storage so that the pipeline can recover from failures."
```
Comment (elfenheart) [inline cdap-ui/app/directives/my-pipeline-configurations/my-realtime-pipeline-config/my-realtime-pipeline-config.html:140]: @bdmogal can we confirm this message? Having double `to` does not seem right

File: cdap-ui/app/directives/my-pipeline-configurations/my-realtime-pipeline-config/my-realtime-pipeline-config.html [hunk 3]
```diff
@@ -14,7 +14,355 @@
...
+        </div>
+        <div class="label-with-toggle instrumentation row">
+          <span class="toggle-label col-xs-4">Instrumentation</span>
+          <div class="col-xs-7 toggle-container">
+            <my-toggle-switch-widget
+              is-on="RealtimePipelineConfigCtrl.instrumentation"
+              on-toggle="RealtimePipelineConfigCtrl.onInstrumentationChange()"
+            ></my-toggle-switch-widget>
+            <i class="fa fa-info-circle"
+                uib-tooltip="Emits timing metrics such as total time, mean, standard deviation for pipeline stages."
```
Comment (elfenheart) [inline cdap-ui/app/directives/my-pipeline-configurations/my-realtime-pipeline-config/my-realtime-pipeline-config.html]: perhaps adds an `and` before `standard deviation`?

File: cdap-ui/app/directives/my-pipeline-scheduler/my-pipeline-scheduler.js [hunk 4]
```diff
@@ -87,18 +93,55 @@ class MyPipelineSchedulerCtrl {
...
+        .then(
+          () => {
+            this.onClose();
+            return this.$state.reload();
+          },
+          (err) => {
+            this.savingSchedule = false;
+            this.myAlertOnValium.show({
+              type: 'danger',
+              content: typeof err === 'object' ? err.data : 'Updating pipeline failed: '+ err
```
Comment (elfenheart) [inline cdap-ui/app/directives/my-pipeline-scheduler/my-pipeline-scheduler.js]: nit: space before `+`

**Review by elfenheart — APPROVED** [review]
Minor Comment, otherwise 👍  LGTM

**Comment by bdmogal** [comment]
@ajainarayanan is editing alerts not part of this PR?

**Comment by bdmogal** [comment]
@ajainarayanan when I change the schedule, the Start Schedule button is disabled. This seems contrary to the behavior in the configure button, where if you change anything, the Apply and Run button is enabled.

**Comment by bdmogal** [comment]
@ajainarayanan can we add some sort of an indicator that the changes are being applied, and the page will now refresh?

**Comment by bdmogal** [comment]
@ajainarayanan did the alignment here change? Shouldn't the left edge of the modeless be aligned with the Configure button?

image

**Comment by bdmogal** [comment]
@ajainarayanan seems like the Apply button changes to Save  in the Advanced configs. Can we just use Save, and Save and Run everywhere?

**Comment by bdmogal** [comment]
@ajainarayanan I don't see any loading state for the Save button. When I click Save and Run, the Start button starts spinning, but it never goes into RUNNING State - the page does not refresh either.

**Comment by bdmogal** [comment]
@ajainarayanan when you click the Save and Start Schedule button, the button becomes really small and the loading icon appears. The size of the button should not change.

**Comment by leacd** [comment]
Hey there,
Here my feedback
- Please  highlight Schedule tab (icon and name) when modeless is open)
- Let's please Include only one main CTA:
The button label needs to be dynamic: if no changes are made to the schedule, it is labeled “start schedule” if edits are implemented it changes to “save and start schedule”.
Per my chat with @bdmogal, if the user changes the schedule, we need to add a "Save" link next to [Save and start schedule] main CTA.
 
image

**Comment by bdmogal** [comment]
@ajainarayanan when I change stagelogging,, the Save button is still disabled.

**Comment by bdmogal** [comment]
@ajainarayanan nothing happens when I change resources and hit Save

**Comment by bdmogal** [comment]
When I click on Save and Run, both buttons start spinning:
image

**Comment by bdmogal** [comment]
@ajainarayanan we need to handle the behavior of Save and Run when the pipeline is already running. I think for simplicity, and because we do not show multiple runs in pipelines today, let's only enable Save and Run when the pipeline is not already running. If it is running, lets disable the Save and Run button, with a tooltip - Use the Save button to Save changes. They will take effect the next time the pipeline is run.

**Review by bdmogal — APPROVED** [review]
Two small comments. Other than that, LGTM

**Comment by bdmogal** [comment]
This also fixes https://issues.cask.co/browse/CDAP-11671

**Comment by ajainarayanan** [comment]
@bdmogal Have added fix for loading issue. Will disable "Save & Run" button if the pipeline is already running in a subsequent improvement PR.

**Pull request merged**